src/operations/query/r4.js

```diff
@@ -101,142 +101,82 @@ class R4SearchQueryCreator {
-                    } else if (args[`${queryParameter}:below`]) {
-                        filterByBelow({
-                            propertyObj, args, queryParameter, columns
-                        }).forEach(q => and_segments.push(negation ? {$nor: [q]} : q));
-                    } else if (args[`${queryParameter}:text`]) {
-                        filterByPartialText({
-                            args, queryParameter, propertyObj, columns,
-                        }).forEach(q => and_segments.push(q));
-                    } else if (args[`${queryParameter}:not:text`] || args[`${queryParameter}:text:not`]) {
-                        filterByPartialText({
-                            args, queryParameter, propertyObj, columns,
-                        }).forEach(q => and_segments.push({$nor: [q]}));
-                    }
+            // set type of field in propertyObj
+            propertyObj.fieldType = this.fhirTypesManager.getTypeForField(
+                {
+                    resourceType,
+                    field: propertyObj.field
+                }
+            );
+            /**
+             * @type {string | string[]}
+             */
+            let queryParameterValue = args[argName];
+            queryParameterValue = convertGraphQLParameters(
+                queryParameterValue,
+                args,
+                queryParameter
+            );
+
+            if (queryParameterValue) {
+
+                let {columns, andSegments} = this.getColumnsAndSegmentsForParameterType({
+                    resourceType, queryParameter, queryParameterValue, propertyObj
+                });
+
+                // replace andSegments according to modifiers
+                if (modifiers.includes('missing')) {
+                    andSegments = filterByMissing({
+                        args, queryParameter, propertyObj, columns
+                    });
+                } else if (modifiers.includes('contains')) {
+                    andSegments = filterByContains({
+                        propertyObj, queryParameterValue, columns
+                    });
+                } else if (modifiers.includes('above')) {
+                    andSegments = filterByAbove({
+                        propertyObj, queryParameterValue, columns
+                    });
+                } else if (modifiers.includes('below')) {
+                    andSegments = filterByBelow({
+                        propertyObj, queryParameterValue, columns
+                    });
+                } else if (modifiers.includes('text')) {
+                    columns = new Set(); // text overrides datatype column logic
+                    andSegments = filterByPartialText({
+                        args, queryParameter, propertyObj, columns,
+                    });
+                }
+
+                // apply negation according to not modifier and add to final collection
+                if (modifiers.includes('not')) {
```

Comment:
instead of using the "negation" variable everywhere, we'll just apply the negative (or not) when we're adding the new andSegments into the totalAndSegment